cmd/notation/verify.go

```diff
@@ -148,3 +171,31 @@ func resolveReference(ctx context.Context, opts *SecureFlagOpts, reference strin
 
 	return ref, nil
 }
+
+func printResult(outputFormat, reference string, outcome *notation.VerificationOutcome) error {
+
+	if reflect.DeepEqual(outcome.VerificationLevel, trustpolicy.LevelSkip) {
+		if outputFormat == ioutil.OutputPlaintext {
+			fmt.Println("Trust policy is configured to skip signature verification for", reference)
+			return nil
+		} else {
+			output := verifyOutput{Reference: reference, Result: "Skipped", UserMetadata: map[string]string{}}
```

Comment:
Looks like when verification is skipped, the two output formats are printing out different messages.
1. Plain text: `Trust policy is configured to skip signature verification for xxx`
2. JSON: `Reference: reference, Result: "Skipped", UserMetadata: map[string]string{}`
IMO, both of them should explicitly state that the trust policy configuration triggered the skip.

Reply:
since the JSON output is meant to be programmatically consumed, we should avoid full messages. would a "Result" value of "TrustPolicySkipped" (or something similar) be ok? suggestions on naming welcome

Reply:
> There is no issue linked with this PR. Could you link at least one issue to this PR so that reviewers can have context?
> 
> Since this is a new feature, could you add E2E tests accordingly?

linked the related metadata PR and issue in the initial comment. will add E2E tests for this PR

Reply:
> since the JSON output is meant to be programmatically consumed, we should avoid full messages. would a "Result" value of "TrustPolicySkipped" (or something similar) be ok? suggestions on naming welcome

IMO, `TrustPolicySkipped` sounds like the trust policy is skipped instead of the verification. How about `SkippedByTrustPolicy`?

Reply:
sounds good to me, will add